### ISSUE TITLE
Type imports are incorrectly reported as unused

### ISSUE THREAD

**Issue by tylerlaprade** [quick-lint/quick-lint-js#1225]
```ts
import type bar from 'bar';
type foo = (typeof bar)[number];
```
results in
`use of undeclared variable: barquick-lint-js[E0057](https://quick-lint-js.com/errors/E0057/)`
